### PR TITLE
Hyprland ecosystem update

### PR DESCRIPTION
**Pull request by JohnMH**
This PR fixes gbm dependencies and adds releases 0.2.0 and 0.3.1 of gui-libs/aquamarine.

Fixes USE="X" dependencies for gui-wm/hyprland-0.42.0 and -9999.

Closes: https://bugs.gentoo.org/936653
Closes: https://bugs.gentoo.org/937638
Closes: https://bugs.gentoo.org/937601

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by JohnMH** [comment]
Per upstream, the simpleWindow test is broken and will soon be removed or rewritten.

**Comment by JohnMH** [comment]
Please note that the there is only one test, the simpleWindow test. Nothing else is skipped as a result.

**Review by MrRoy — APPROVED** [review]

File: gui-wm/hyprland/hyprland-0.42.0-r1.ebuild [hunk 1]
```diff
@@ -46,25 +57,22 @@ RDEPEND="
...
 	>=gui-libs/hyprutils-0.2.1
 "
 BDEPEND="
-	${WLROOTS_BDEPEND}
 	|| ( >=sys-devel/gcc-13:* >=sys-devel/clang-16:* )
 	app-misc/jq
 	dev-build/cmake
 	>=dev-util/hyprwayland-scanner-0.3.8
 	virtual/pkgconfig
+	${XWAYLAND_DEPEND}
```
Comment (dlan17) [inline gui-wm/hyprland/hyprland-0.42.0-r1.ebuild:75]: I think ${XWAYLAND_DEPEND} should be removed from BDEPEND.

BDEPEND specifies dependencies applicable to CBUILD, usually BDEPEND won't contribute to link/run time phase.

https://devmanual.gentoo.org/general-concepts/dependencies/index.html#build-dependencies

File: gui-libs/aquamarine/aquamarine-0.1.1-r2.ebuild [hunk 2]
```diff
@@ -19,6 +19,8 @@ fi
 LICENSE="BSD"
 SLOT="0"
 
+# Upstream states that the simpleWindow test is broken.
```
Comment (dlan17) [inline gui-libs/aquamarine/aquamarine-0.1.1-r2.ebuild:22]: To better track this in the future  for other maintainers
I'd suggest also append 'bug 936653' to the end of comment, also any URL link from upstream?

File: gui-wm/hyprland/hyprland-0.42.0-r1.ebuild [hunk 3]
```diff
@@ -32,6 +32,17 @@ HYPRPM_RDEPEND="
 	dev-vcs/git
 	virtual/pkgconfig
 "
+
+XWAYLAND_DEPEND="
+	X? (
+		x11-libs/libxcb:0=
+		x11-base/xwayland
+		x11-libs/xcb-util-errors
+		x11-libs/xcb-util-renderutil
```
Comment (TheAifam5) [inline gui-wm/hyprland/hyprland-0.42.0-r1.ebuild:41]: I don't think `x11-libs/xcb-util-renderutil` is needed. I don't have it even installed on my setup.

File: gui-wm/hyprland/hyprland-9999.ebuild [hunk 4]
```diff
@@ -32,6 +32,17 @@ HYPRPM_RDEPEND="
 	dev-vcs/git
 	virtual/pkgconfig
 "
+
+XWAYLAND_DEPEND="
+	X? (
+		x11-libs/libxcb:0=
+		x11-base/xwayland
+		x11-libs/xcb-util-errors
+		x11-libs/xcb-util-renderutil
```
Comment (TheAifam5) [inline gui-wm/hyprland/hyprland-9999.ebuild:41]: x11-libs/xcb-util-renderutil is not needed.

Comment (JohnMH) [reply gui-wm/hyprland/hyprland-9999.ebuild:41]: I'm running a clean build to confirm that.

**Pull request closed without merging**